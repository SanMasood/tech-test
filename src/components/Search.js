import React, {useState} from 'react';

import getImages from '../requests/getImages';
import '../styles/Search.css'
import logoImage from '../assets/Nasa-logo-3D.gif'



const Search = () => {
    const [value, setValue] = useState('');

    const handleSubmit= (event) => {
        event.preventDefault();
        getImages(value);
    }

return (
    
    <div className="Search">
        <img src={logoImage} data-testid="logo-id" className="nasa-logo" alt="NASAlogo"/>

        <form className="input-form" onSubmit={handleSubmit}>Search: 
        <input className="search-text" type="text" onChange={(e)=>setValue(e.target.value)}/>

        <button type="submit" className="search-button">Go!</button>

        </form>
    </div>
)
}

export default Search;