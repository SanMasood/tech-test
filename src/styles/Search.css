.Search{
    display: flex;
    flex-direction: row;
    align-content: center;
    justify-content: center;
    height: 20px;
}


.search-text{
    width: 50%;
    height: 20px;
    color: rgb(29, 3, 82);
    border-radius: 2px;
    border: 1px solid grey;
    background-color: rgb(236, 236, 236);   
    margin: 5px; 
    padding: 0 10px;
    
}
.search-button{
    height: 22px;
    border: 1px solid rgb(28, 28, 104);
    border-radius: 2px;
}

