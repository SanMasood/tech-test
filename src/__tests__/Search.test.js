import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Search from '../components/Search';


describe ("Search", () => {
    it("renders correctly", () => {
    
      const {asFragment} = render (
        <Search         
        />
      )
      expect(asFragment()).toMatchSnapshot();  
    });


    it('Handles button submit', () => {

      const { getByLabelText, asFragment } = render(<Search />);
      const searchQuery = screen.getByLabelText('Search:');
      userEvent.type(searchQuery, 'moon');

      const submitButton = screen.getByRole('button', { name: /Go/i });
      userEvent.click(submitButton);

    });

  
    it('calls "onClick" prop on button click', () => {

      const onSubmit = jest.fn();
      const { getByRole } = render(<button onClick={onSubmit}  />);
    
      fireEvent.click(getByRole('button'));
      expect(onSubmit).toHaveBeenCalled();
    });   
     
});
