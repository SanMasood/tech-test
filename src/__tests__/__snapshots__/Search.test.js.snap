// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly 1`] = `
<DocumentFragment>
  <div
    class="Search"
  >
    <form
      class="input-form"
    >
      Search:
      <input
        class="search-text"
        type="text"
      />
      <button
        class="search-button"
        type="submit"
      >
        Go!
      </button>
    </form>
  </div>
</DocumentFragment>
`;
